opentracing-spring-cloud-starter/src/test/java/io/opentracing/contrib/spring/cloud/feign/FeignTest.java

```diff
@@ -0,0 +1,69 @@
+package io.opentracing.contrib.spring.cloud.feign;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.netflix.feign.EnableFeignClients;
+import org.springframework.cloud.netflix.feign.FeignClient;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.opentracing.Tracer;
+import io.opentracing.mock.MockTracer;
+import io.opentracing.util.ThreadLocalActiveSpanSource;
+
+/**
+ * @author Pavol Loffay
+ */
+@SpringBootTest(
+        webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT,
+        classes = {FeignTest.FeignWithoutRibbonConfiguration.class})
+@TestPropertySource(properties = {"server.port=13598", "endpoints.health.enabled=true"})
```

Comment:
Can't you use a random port? Possibly not, because of Feign, but I'm asking anyway :)

Reply:
you are right, I wish we could. Random port can be injected but it does not work because feign requires it at compile time